packages/flutter_tools/templates/plugin_ffi/android.tmpl/build.gradle.tmpl

```diff
@@ -0,0 +1,57 @@
+// The Android Gradle Plugin builds the native code with the Android NDK.
+
+group '{{androidIdentifier}}'
+version '1.0'
+
+buildscript {
+    repositories {
+        google()
+        mavenCentral()
+    }
+
+    dependencies {
+        // The Android Gradle Plugin knows how to build native code with the NDK.
+        classpath 'com.android.tools.build:gradle:4.1.0'
+    }
+}
+
+rootProject.allprojects {
+    repositories {
+        google()
+        mavenCentral()
+    }
+}
+
+apply plugin: 'com.android.library'
+
+android {
+    compileSdkVersion 31
+
+    // Invoke the shared CMake build with the Android Gradle Plugin.
+    //
+    // We do _not_ specify the NDK and CMake version here. This way we will
+    // automatically get the default versions from the Android Gradle Plugin.
+    //
+    // Android Gradle Plugin 4.1.0: Android NDK 21.1.6352462.
+    // https://developer.android.com/studio/projects/install-ndk#default-ndk-per-agp
+    // CMake 3.10.2.
+    // https://developer.android.com/studio/projects/install-ndk#vanilla_cmake
+    //
+    // You are strongly encouraged to not ship FFI plugins requiring newer
+    // versions of the NDK or CMake, because your clients will not be able to
+    // use them.
```

Comment:
I left a comment in the doc related to this part.  

The problem is that in Flutter the Gradle/AGP versions to be used are actually specified in the app instead of the plugin. In fact, the AGP/Gradle version doesn't have any effect unless the plugin is built as aar with `flutter build aar`.

Reply:
I have been able to verify this behavior. The `android.ndkVersion` in the FFI plugin `build.gradle` is ignored when building a Flutter app that imports it. The `android.ndkVersion` in the apps `app/build.gradle` is used.

We'll need to find a solution for this behavior.